runtime/src/main/java/org/corfudb/runtime/object/AlwaysLatestSnapshot.java

```diff
@@ -0,0 +1,30 @@
+package org.corfudb.runtime.object;
+
+import lombok.NonNull;
+import org.rocksdb.OptimisticTransactionDB;
+
+/**
+ * Provides read-committed consistency guarantees.
+ *
+ * @param <S> extends SnapshotGenerator
+ */
+public class AlwaysLatestSnapshot<S extends SnapshotGenerator<S>> implements SMRSnapshot<S> {
+    private final OptimisticTransactionDB rocksDb;
+    private final ViewGenerator<S> viewGenerator;
+
+    public AlwaysLatestSnapshot(@NonNull OptimisticTransactionDB rocksDb,
+                                @NonNull ViewGenerator<S> viewGenerator) {
+        this.rocksDb = rocksDb;
+        this.viewGenerator = viewGenerator;
+    }
+
+    @Override
+    public S consume() {
+        return viewGenerator.newView(new RocksDbReadCommittedTx(rocksDb));
```

Comment:
This looks like the transactional behaviour is changing for read committed between the VLO and MVO.

In the case of the MVO, imagine T1 begins a transaction but does not perform reads right away. Then T2 begins a transaction and does a read which triggers a sync. This sync then begins to load CP entries with multiple continuation entries. After the records from the first continuation entry are applied but before the second continuation entry is processed, T1 now does reads for two keys that are in the table, but because the whole checkpoint is not yet applied, only observes one of them and goes to commit. In this case, T1 observed state that does not correspond to a valid transactional boundary.

In the case of the VLO, all reads had to go through the stamped lock, so if T2 was performing a sync, the reads for T1 were blocked until T2 released the lock.